src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets

```diff
@@ -99,7 +99,7 @@
     <!-- add relative blob path metadata -->
     <ItemGroup>
       <ForUpload>
-        <RelativeBlobPath>$(Platform)$(ConfigurationGroup)/Tests/%(Filename)%(Extension)</RelativeBlobPath>
+        <RelativeBlobPath>$(Platform)$(ConfigurationGroup)/Tests/$([System.String]::Copy('%(RecursiveDir)').Replace('\', '/'))%(Filename)%(Extension)</RelativeBlobPath>
```

Comment:
It's one of the oddities of msbuild (maybe not that odd if you don't know C I guess); it does work as intended.